src/Common/tests/System/Net/Http/LoopbackServer.cs

```diff
@@ -742,40 +743,61 @@ public override async Task SendResponseAsync(HttpStatusCode statusCode = HttpSta
                 string headerString = null;
                 int contentLength = -1;
                 bool isChunked = false;
+                bool hasDate = false;
+                bool hasContentLength  = false;
 
                 if (headers != null)
                 {
+                    // Process given headers and look for some well-known cases.
                     foreach (HttpHeaderData headerData in headers)
                     {
-                        headerString = headerString + $"{headerData.Name}: {headerData.Value}\r\n";
                         if (headerData.Name.Equals("Content-Length", StringComparison.OrdinalIgnoreCase))
                         {
+                            hasContentLength = true;
```

Comment:
```
hasContentLength = headerData.Name.Equals("Content-Length", StringComparison.OrdinalIgnoreCase)
if (hasContentLength)
```

assuming you cannot have more than one content-length - if you cannot it might also be worth to add an assert against more than 1 occurrence. Disregard if it's allowed

Reply:
would it make sense to actually convert contentLength to `Nullable<int>`?